client/ayon_core/hosts/nuke/api/utils.py

```diff
@@ -142,3 +149,160 @@ def is_headless():
+
+
+def create_error_report(context):
+    error_message = ""
+    success = True
+    for result in context.data["results"]:
+        if result["success"]:
+            continue
+
+        success = False
+
+        err = result["error"]
+        formatted_traceback = "".join(
+            traceback.format_exception(
+                type(err),
+                err,
+                err.__traceback__
+            )
+        )
+        fname = result["plugin"].__module__
+        if 'File "<string>", line' in formatted_traceback:
+            _, lineno, func, msg = err.traceback
+            fname = os.path.abspath(fname)
+            formatted_traceback = formatted_traceback.replace(
+                'File "<string>", line',
+                'File "{0}", line'.format(fname)
+            )
+
+        err = result["error"]
+        error_message += "\n"
+        error_message += formatted_traceback
+
+    return success, error_message
+
+
+def submit_headless_farm(node):
+    # Ensure code is executed in root context.
+    if nuke.root() == nuke.thisNode():
+        _submit_headless_farm(node)
+    else:
+        # If not in root context, move to the root context and then execute the
+        # code.
+        with nuke.root():
+            _submit_headless_farm(node)
+
+
+def _submit_headless_farm(node):
+    context = util.collect()
+
+    success, error_report = create_error_report(context)
+
+    if not success:
+        show_message_dialog(
+            "Collection Errors", error_report, level="critical"
+        )
+        return
+
+    # Find instance for node and workfile.
+    instance = None
+    instance_workfile = None
+    indexes_to_remove = []
+    for count, Instance in enumerate(context):
+        if Instance.data["family"] == "workfile":
+            instance_workfile = Instance
+            continue
+
+        instance_node = Instance.data["transientData"]["node"]
+        if node.name() == instance_node.name():
+            instance = Instance
+        else:
+            indexes_to_remove.append(count)
+
+    if instance is None:
+        show_message_dialog(
+            "Collection Error",
+            "Could not find the instance from the node.",
+            level="critical"
+        )
+        return
+
+    # Enable for farm publishing.
+    instance.data["farm"] = True
+    instance.data["transfer"] = False
+
+    # Clear the families as we only want the main family, ei. no review etc.
+    instance.data["families"] = []
+
+    # Use the workfile instead of published.
+    publish_attributes = instance.data["publish_attributes"]
+    publish_attributes["NukeSubmitDeadline"]["use_published_workfile"] = False
+
+    # Disable version validation.
+    instance.data.pop("latestVersion")
+    instance_workfile.data.pop("latestVersion")
+
+    # Remove all other instances.
+    indexes_to_remove.sort(reverse=True)
+    for i in indexes_to_remove:
+        if 0 <= i < len(context):
+            del context[i]
```

Comment:
The is in range change would only be obscuring an actual bug - since you're removing from the back of the indices that should never occur. And if it does, then you have bigger issues to worry about because it means your indices would be misaligned.
```suggestion
    # Remove all other instances.
    for i in reversed(indexes_to_remove):
        del context[i]
```

Reply:
Not sure where you got this from, but this code was removed (on your request :) )

Reply:
Interesting - github was still showing it to me somehow. Ignore me then.